src/cache-fs.ts

```diff
@@ -97,9 +147,54 @@ export class CacheFs implements FileSystem {
         });
     }
 
+    private rescanOnError() {
+        this.cache.loadDirectoryTree().then(cachedTree => {
+            this.isTreeCached = false;
+            this.loadDirectoryTree().then(realTree => {
+                const {toDelete, toAdd, toChange} = getTreesDiff(
+                    nodesToMap(cachedTree.children),
+                    nodesToMap(realTree.children)
+                );
+
```

Comment:
@amir-arad if you agree that this should be done in separate issue later, then re-review PR please.